worker/src/loader/file_loader.py

```diff
@@ -0,0 +1,63 @@
+from collections.abc import Iterator
+from datetime import UTC, datetime
+from typing import TYPE_CHECKING
+
+from langchain_core.document_loaders import BaseLoader
+from langchain_core.documents import Document
+from unstructured.chunking.title import chunk_by_title
+from unstructured.partition.auto import partition
+
+from redbox.models.file import File
+from redbox.models.settings import Settings
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3.client import S3Client
+else:
+    S3Client = object
+
+
+class UnstructuredDocumentLoader(BaseLoader):
+    """Load, partition and chunk a document using local unstructured library"""
+
+    def __init__(self, file: File, s3_client: S3Client, env: Settings) -> None:
+        """Initialize the loader with a file path.
+
+        Args:
+            file: The RedBox File to load
+        """
+        self.file = file
+        self.s3_client = s3_client
+        self.env = env
+
+    def lazy_load(self) -> Iterator[Document]:  # <-- Does not take any arguments
+        """A lazy loader that reads a file line by line.
+
+        When you're implementing lazy load methods, you should use a generator
```

Comment:
but you dont?

Reply:
Good point :)  I used the DocumentLoader structure but actually the next step in the runnable wants a list so it can batch things. Cleanest way would be to collect the generator in a runnable between the two. I'll sort it in the next PR for this?

Reply:
Fixed here because mypy was whinging